src/agent/srcview/src/bin/srcview.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use anyhow::{format_err, Context, Result};
+use srcview::{ModOff, Report, SrcLine, SrcView};
+use std::fs::{self};
+use std::io::{stdout, Write};
+use std::path::{Path, PathBuf};
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+enum Opt {
+    Srcloc(SrcLocOpt),
+    PdbPaths(PdbPathsOpt),
+    Cobertuna(CobertunaOpt),
+    /// Print 3rd-party license information
+    Licenses,
+}
+
+/// Print the file paths in the provided PDB
+#[derive(StructOpt, Debug)]
+struct PdbPathsOpt {
+    pdb_path: PathBuf,
+}
+
+/// Print modoffset file with file and source lines
+#[derive(StructOpt, Debug)]
+struct SrcLocOpt {
+    pdb_path: PathBuf,
+    modoff_path: PathBuf,
+    #[structopt(long)]
+    module_name: Option<String>,
+}
+
+/// Generate a Cobertuna XML coverage report
+///
+/// Example:
+///   srcview cobertuna ./res/example.pdb res/example.txt
+///             --include-regex "E:\\\\1f\\\\coverage\\\\"
+///             --filter-regex "E:\\\\1f\\\\coverage\\\\"
+///             --module-name example.exe
+///
+/// In this example, only files that live in E:\1f\coverage are included and
```

Comment:
nit: should we note that as these are regex's, backslashes must be double escaped (as this example does) to prevent them from being treated as backreferences?

Reply:
@jopletchMSFT, you've nailed the reasoning.

The regular expression to match `e:\foo` is `e:\\foo`, as a single backslash is a configuration marker for regular expressions.  This can include back references, in the case of `\1`, or things like "any white space" in the case of `\s`.

Since this is coming from the shell, we also need to escape the shell's interpolation of \.

In practice:
* providing `e:\1f\coverage`  gives the regex engine "e:1fcoverage".
* providing `e:\\1f\\coverage` gives the regex engine  `e:\1f\coverage`, which interprets  `\1` is an undefined back reference.

To have what we want, a regex that matches `e:\1f\coverage`, you must provide `e:\\\\1f\\\\coverage`

Reply:
This is why I suggested investigating potentially reusing the coverage mechanism for selecting modules/etc.